### PR TITLE
add ani-cli.profile

### PR DESCRIPTION
**Pull request by pirate486743186**
shell script for streaming anime

### CODE REVIEW

**Comment by kmk3** [comment]
(Rebased to master to fix CI)

Just to confirm, is this <https://github.com/pystardust/ani-cli>?

File: etc/profile-a-l/ani-cli.profile [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Firejail profile for ani-cli
+# Description: Shell script to watch Anime from the terminal
+# This file is overwritten after every install/update
+quiet
+# Persistent local customizations
+include ani-cli.local
+# Persistent global definitions
+include globals.local
+
+noblacklist ${DOWNLOADS}
```
Comment (glitsj16) [inline etc/profile-a-l/ani-cli.profile:10]: Drop this line. ${DOWNLOADS} is never blacklisted by disable-xdg.inc (as is ${DESKTOP}.

File: etc/profile-a-l/ani-cli.profile [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-proc.inc
+include disable-programs.inc
+include disable-shell.inc
+include disable-xdg.inc
+
+mkdir ${HOME}/.cache/ani-cli
+mkdir ${HOME}/.local/state/ani-cli
+
```
Comment (glitsj16) [inline etc/profile-a-l/ani-cli.profile:31]: Drop this empty line.

File: etc/profile-a-l/ani-cli.profile [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+whitelist ${DOWNLOADS}
+whitelist ${HOME}/.cache/ani-cli
+whitelist ${HOME}/.config/mpv
+whitelist ${HOME}/.local/state/ani-cli
+
```
Comment (glitsj16) [inline etc/profile-a-l/ani-cli.profile:36]: Drop this empty line.

File: etc/profile-a-l/ani-cli.profile [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+netfilter
+nodvd
+nogroups
+noinput
+nonewprivs
+noprinters
+noroot
+notv
+nou2f
+protocol unix,inet,inet6,netlink
```
Comment (glitsj16) [inline etc/profile-a-l/ani-cli.profile:55]: Does it work without netlink?

File: etc/profile-a-l/ani-cli.profile [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+private-dev
+private-etc alsa,alternatives,asound.conf,ca-certificates,crypto-policies,fonts,host.conf,hostname,hosts,ld.so.cache,ld.so.conf,ld.so.conf.d,ld.so.preload,locale,locale.alias,locale.conf,localtime,machine-id,mime.types,nsswitch.conf,pango,pki,protocols,pulse,resolv.conf,rpc,services,ssl,X11,xdg
+private-tmp
+
+dbus-user none
+dbus-system none
+
+read-only ${HOME}/.config/mpv
+
+restrict-namespaces
```
Comment (glitsj16) [inline etc/profile-a-l/ani-cli.profile:72]: As this is a CLI app, can we add memory-deny-write-execute?

File: etc/profile-a-l/ani-cli.profile [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+protocol unix,inet,inet6,netlink
+seccomp
+seccomp.block-secondary
+tracelog
+
+disable-mnt
+private-bin ani-cli,aria2c,cat,cp,curl,cut,ffmpeg,fzf,grep,head,mkdir,mpv,mv,nl,sed,sh,sort,tput,tr,uname,wc
+private-cache
+private-dev
+private-etc alsa,alternatives,asound.conf,ca-certificates,crypto-policies,fonts,host.conf,hostname,hosts,ld.so.cache,ld.so.conf,ld.so.conf.d,ld.so.preload,locale,locale.alias,locale.conf,localtime,machine-id,mime.types,nsswitch.conf,pango,pki,protocols,pulse,resolv.conf,rpc,services,ssl,X11,xdg
```
Comment (glitsj16) [inline etc/profile-a-l/ani-cli.profile:64]: There's a rework happening for private-etc (cfr. #5610). I suggest you keep this 'old' format commented and add the 'new' format just below it. That would be `private-etc @network,@sound,@tls-ca,@x11,mime.types`.

**Review by glitsj16 — CHANGES_REQUESTED** [review]
Some checks needed for potential hardenings.

File: etc/profile-a-l/ani-cli.profile (hunk 6 above)
Comment (kmk3) [reply etc/profile-a-l/ani-cli.profile:64]: Same note as in the following comment:

* <https://github.com/netblue30/firejail/pull/5706#discussion_r1124284952>

Comment (glitsj16) [reply etc/profile-a-l/ani-cli.profile:64]: Got it! Same reply as in https://github.com/netblue30/firejail/pull/5706#discussion_r1124297337.

**Review by glitsj16 — CHANGES_REQUESTED** [review]
Some nits, but removing the `noblacklist ${DOWNLOADS}` is the main remark I have.

**Review by glitsj16 — APPROVED** [review]
LGTM

File: etc/profile-a-l/ani-cli.profile [hunk 7]
```diff
@@ -0,0 +1,41 @@
+# Firejail profile for ani-cli
+# Description: Shell script to watch Anime from the terminal
+# This file is overwritten after every install/update
+quiet
+# Persistent local customizations
+include ani-cli.local
+# Persistent global definitions
```
Comment (glitsj16) [inline etc/profile-a-l/ani-cli.profile:7]: Now that this is a redirect profile, it's important to `avoid including globals.local twice`. Here's what we use in other profiles for that purpose:
```
# added by included profile
#include globals.local
```

**Review by glitsj16 — CHANGES_REQUESTED** [review]
Avoid including globals.local twice. Format to do that is included in my comment.

File: etc/profile-a-l/ani-cli.profile [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+disable-mnt
+private-bin ani-cli,aria2c,cat,cp,curl,cut,ffmpeg,fzf,grep,head,mkdir,mv,nl,sed,sh,sort,tput,tr,uname,wc
+#private-cache
+private-etc alsa,alternatives,asound.conf,ca-certificates,crypto-policies,fonts,host.conf,hostname,hosts,ld.so.cache,ld.so.conf,ld.so.conf.d,ld.so.preload,locale,locale.alias,locale.conf,localtime,machine-id,mime.types,nsswitch.conf,pango,pki,protocols,pulse,resolv.conf,rpc,services,ssl,X11,xdg
+private-tmp
+
+read-only ${HOME}/.config/mpv
+
+# Redirect
+include mpv.profile
```
Comment (rusty-snake) [inline etc/profile-a-l/ani-cli.profile:42]: mpv is just the default. Other player like vls are supported to. Redirecting it to mpv makes it harder for users to create locals that allow other players.

Comment (rusty-snake) [inline etc/profile-a-l/ani-cli.profile:42]: Does it need access to Downloads, Documents, ...?

**Review by rusty-snake — COMMENTED** [review]
I'm sceptical about the mpv redirect, otherwise LGTM.

File: etc/profile-a-l/ani-cli.profile (hunk 8 above)
Comment (glitsj16) [reply etc/profile-a-l/ani-cli.profile:42]: I can understand the skepticism. It's a bit similar to Firejail's favoritism towards Firefox, which is very understandable from the history of the project. Think fireurl (regardless of implementation) with multi web-browser functionality. On occasion I've been playing with the idea to write such a meta-web-browser.profile that would tackle these kind of 'issues'. Nothing that's very concrete yet though. In a similar vain we might consider a meta-media-player.profile that supports whatever media-player applications Firejail currently supports.

Totally out of scope for the currently opened PR's from @pirate486743186 IMO. But something to have a wider discussion on at some point?

Comment (glitsj16) [reply etc/profile-a-l/ani-cli.profile:41]: @pirate486743186 @rusty-snake Any thoughts on #5712 to unbreak mpv lock-in?

Comment (kmk3) [reply etc/profile-a-l/ani-cli.profile:41]: @glitsj16 [on Mar 7](https://github.com/netblue30/firejail/pull/5707#discussion_r1127767622):

> Any thoughts on #5712 to unbreak mpv lock-in?

I have something half-written up; I'll post it in #5712 later.

Comment (glitsj16) [reply etc/profile-a-l/ani-cli.profile:41]: @kmk3 No rush. I just wanted to avoid the topic getting snowed under. Looking forward to reading your views!

**Comment by netblue30** [comment]
all in, thanks!

**Pull request merged**